packages/shared/src/features/ingestion/types.ts

```diff
@@ -145,15 +145,44 @@ export const UpdateGenerationBody = UpdateSpanBody.extend({
   return false;
 });
 
-export const ScoreBody = z.object({
+const BaseScoreBody = z.object({
   id: z.string().nullish(),
   name: NonEmptyString,
-  value: z.number(),
   traceId: z.string(),
   observationId: z.string().nullish(),
   comment: z.string().nullish(),
+  configId: z.string().nullish(),
 });
 
+export const ScoreBody = z.discriminatedUnion("dataType", [
+  BaseScoreBody.merge(
+    z.object({
+      value: z.number(),
```

Comment:
I'd add custom zod errors here as this might be the biggest issue when folks use the api, e.g. "value must be numeric for numeric scores"